pkg/kubectl/bind-apiservice/plugin/servicebindings.go

```diff
@@ -139,13 +141,156 @@ func (b *BindAPIServiceOptions) createAPIServiceBindings(ctx context.Context, co
+	var err error
+
+	donate := false
+	if claim.Create != nil {
+		donate = claim.Create.Donate
+	}
+	adopt := claim.Adopt
+
+	name := ""
+	var owner kubebindv1alpha1.Owner
+	if (claim.Selector != kubebindv1alpha1.ResourceSelector{}) {
+		name = claim.Selector.Name
+		owner = claim.Selector.Owner
+	}
+	var postamble string
+	switch {
+	case !donate && !adopt:
+		groupResource = "read " + groupResource
+	case donate && !adopt:
+		groupResource = "create " + groupResource
+	case !donate && adopt:
+		groupResource = "have ownership of " + groupResource
+		postamble = " and assign the ownership to you"
+	case donate && adopt:
+		groupResource = "create" + groupResource
+		postamble = " and wants to keep the ownership"
+	}
+
+	_, err = fmt.Fprintf(b, "The provider wants to %s on your cluster%s.", groupResource, postamble) //xxx
+
+	if owner == kubebindv1alpha1.Consumer {
+		owner = "you"
+	}
+	if owner == kubebindv1alpha1.Provider {
+		owner = "the provider"
+	}
+	switch {
+	case owner == "" && name == "":
+		_, err = fmt.Fprintf(b, "\n")
+	case owner != "" && name == "":
+		_, err = fmt.Fprintf(b, " This only applies to objects which are owned by %s.\n", owner)
+	case owner == "" && name != "":
+		_, err = fmt.Fprintf(b, " This only applies to objects which are referenced with:\n	name: \"%s\"\n", name)
+	case owner != "" && name != "":
+		_, err = fmt.Fprintf(b, " This only applies to objects which are owned by %s and to objects which are referenced with:\n	name: \"%s\"\n", owner, name)
+	}
+
+	return err
+
+}
+
+func writeOnConflict(b *bytes.Buffer, claim kubebindv1alpha1.PermissionClaim) error {
+	var err error
+
+	if claim.OnConflict != nil {
+		switch {
+		case claim.OnConflict.ProviderOverwrites && claim.OnConflict.RecreateWhenConsumerSideDeleted:
+			_, err = b.WriteString("Conflicting objects will be overwritten and created objects will be recreated upon deletion.\n")
+		case claim.OnConflict.ProviderOverwrites:
+			_, err = b.WriteString("Conflicting objects will be overwritten and created objects will not be recreated upon deletion.\n")
+		case claim.OnConflict.RecreateWhenConsumerSideDeleted:
+			_, err = b.WriteString("Conflicting objects will not be overwritten and created objects will be recreated upon deletion.\n")
+		default: //Do nothing
+		}
+	}
+	return err
+}
+
+func writeUpdateClause(b *bytes.Buffer, claim kubebindv1alpha1.PermissionClaim) error {
+	var err error
+
+	if claim.Update == nil {
+		return nil
+	}
+
+	if claim.Update.Fields != nil {
+		_, err = b.WriteString("The following fields of the objects will still be able to be changed by owner of this claim:\n")
+	}
+	if claim.Update.Preserving != nil {
+		_, err = b.WriteString("The following fields of the objects will be overwritten with their initial values, if they are modified:\n")
+	}
+
+	for _, s := range append(claim.Update.Fields, claim.Update.Preserving...) {
+		_, err = fmt.Fprintf(b, "	\"%s\"\n", s)
```

Comment:
Is this a tab? If yes, can you use `\t` instead?

Reply:
I changed all "tabs" to `\t` in b9eac5dffe6fa930fd8ef4f2db5ead3c8cb1918f.